api/service/syncing/syncing.go

```diff
@@ -745,28 +724,24 @@ func (ss *StateSync) SyncLoop(bc *core.BlockChain, worker *worker.Worker, isBeac
 	if !isBeacon {
 		ss.RegisterNodeInfo()
 	}
-	// remove SyncLoopFrequency
-	ticker := time.NewTicker(SyncLoopFrequency * time.Second)
 Loop:
 	for {
-		select {
-		case <-ticker.C:
-			otherHeight := ss.getMaxPeerHeight(isBeacon)
-			currentHeight := bc.CurrentBlock().NumberU64()
-			if currentHeight >= otherHeight {
-				utils.Logger().Info().Msgf("[SYNC] Node is now IN SYNC! (isBeacon: %t, ShardID: %d, otherHeight: %d, currentHeight: %d)", isBeacon, bc.ShardID(), otherHeight, currentHeight)
-				break Loop
-			} else {
-				utils.Logger().Debug().Msgf("[SYNC] Node is Not in Sync (isBeacon: %t, ShardID: %d, otherHeight: %d, currentHeight: %d)", isBeacon, bc.ShardID(), otherHeight, currentHeight)
-			}
-			startHash := bc.CurrentBlock().Hash()
-			size := uint32(otherHeight - currentHeight)
-			if size > BatchSize {
-				size = BatchSize
-			}
-			ss.ProcessStateSync(startHash[:], size, bc, worker)
-			ss.purgeOldBlocksFromCache()
+		otherHeight := ss.getMaxPeerHeight(isBeacon)
+		currentHeight := bc.CurrentBlock().NumberU64()
+		if currentHeight >= otherHeight {
+			utils.Logger().Info().Msgf("[SYNC] Node is now IN SYNC! (isBeacon: %t, ShardID: %d, otherHeight: %d, currentHeight: %d)", isBeacon, bc.ShardID(), otherHeight, currentHeight)
+			break Loop
+		} else {
+			utils.Logger().Debug().Msgf("[SYNC] Node is Not in Sync (isBeacon: %t, ShardID: %d, otherHeight: %d, currentHeight: %d)", isBeacon, bc.ShardID(), otherHeight, currentHeight)
+		}
+		startHash := bc.CurrentBlock().Hash()
+		size := uint32(otherHeight - currentHeight)
+		if size > BatchSize {
+			size = BatchSize
 		}
+		ss.ProcessStateSync(startHash[:], size, bc, worker)
+		ss.purgeOldBlocksFromCache()
+		time.Sleep(SyncLoopFrequency * time.Second)
```

Comment:
Who is caller of this go routine, is it necessary to be explicitly blocking

Reply:
ditto: yes, it need to be blocking here. there should be a single state syncing happening as a batch-sequential sync loop